### PR TITLE
Gives spend guardian injectors an appropriate examine text + refactor

### LINKED ISSUE #9903: Holoparasite injectors don't show that they are used upon examination
**Problem Description**:
Holoparasite injectors don't show that they are used upon examination.

**What did you expect to happen**: 
It saying something about being used.

**What happened instead**:
The examine text doesn't tell you anything.

**Why is this bad/What are the consequences:**
As security you don't know if it's used until you use it.

**Steps to reproduce the problem**:
1. Examine an used holoparasite injector

### PR DESCRIPTION
**Pull request by farie82**
**What does this PR do:**
Guardian injectors will now say if they are spend or now. 
And I've done some refactoring
Also removed double announcing the bio guardian to the player. And some other minor things.

Examine text:
image

fixes #9903 

**Changelog:**
:cl: Farie82
tweak: Guardian injectors now tell in their examination text if they are spend or not.
/:cl:

### CODE REVIEW

File: code/game/gamemodes/miniantags/guardian/guardian.dm [hunk 1]
```diff
@@ -435,6 +406,34 @@
 	used_message = "The cluster already hatched."
 	failure_message = "<B>...but soon settles again. Guess they weren't ready to hatch after all.</B>"
 
+/obj/item/guardiancreator/biological/create_theme(var/mob/living/simple_animal/hostile/guardian/G, mob/living/user)
```
Comment (Birdtalon) [inline code/game/gamemodes/miniantags/guardian/guardian.dm]: `var/` is implied in parameters

File: code/game/gamemodes/miniantags/guardian/guardian.dm [hunk 2]
```diff
@@ -435,6 +406,34 @@
...
+	var/bio_list = list("Rose" = "#F62C6B", \
```
Comment (Birdtalon) [inline code/game/gamemodes/miniantags/guardian/guardian.dm]: As above.

File: code/game/gamemodes/miniantags/guardian/guardian.dm [hunk 3]
```diff
@@ -418,6 +363,32 @@
 	failure_message = "<B>...ERROR. BOOT SEQUENCE ABORTED. AI FAILED TO INTIALIZE. PLEASE CONTACT SUPPORT OR TRY AGAIN LATER.</B>"
 	ling_failure = "The holoparasites recoil in horror. They want nothing to do with a creature like you."
 
+/obj/item/guardiancreator/tech/create_theme(var/mob/living/simple_animal/hostile/guardian/G, mob/living/user)
```
Comment (Birdtalon) [inline code/game/gamemodes/miniantags/guardian/guardian.dm]: `var/` is implied in parameters

File: code/game/gamemodes/miniantags/guardian/guardian.dm [hunk 4]
```diff
@@ -418,6 +363,32 @@
...
+	var/tech_list = list("Rose" = "#F62C6B", \
```
Comment (Birdtalon) [inline code/game/gamemodes/miniantags/guardian/guardian.dm]: Same here with the `\`

File: code/game/gamemodes/miniantags/guardian/guardian.dm [hunk 5]
```diff
@@ -325,83 +326,27 @@
 	user.verbs += /mob/living/proc/guardian_recall
 	user.verbs += /mob/living/proc/guardian_reset
 
-	var/color
+	create_theme(G, user)
+
+/obj/item/guardiancreator/proc/create_theme(var/mob/living/simple_animal/hostile/guardian/G, mob/living/user)
```
Comment (Birdtalon) [inline code/game/gamemodes/miniantags/guardian/guardian.dm]: `var/` implied.

File: code/game/gamemodes/miniantags/guardian/guardian.dm [hunk 6]
```diff
@@ -325,83 +326,27 @@
...
-			G.name = "[color] [picked_name]"
-			G.real_name = "[color] [picked_name]"
-			G.icon_living = "[theme][color]"
-			G.icon_state = "[theme][color]"
-			G.icon_dead = "[theme][color]"
-
-			to_chat(user, "[G.bio_fluff_string].")
-			G.attacktext = "swarms"
-			G.speak_emote = list("chitters")
+	to_chat(user, "[G.magic_fluff_string].")
```
Comment (SamHPurp) [inline code/game/gamemodes/miniantags/guardian/guardian.dm]: is this debug text 👀

File: code/game/gamemodes/miniantags/guardian/guardian.dm [hunk 7]
```diff
@@ -418,6 +363,32 @@
...
+	G.name_color = tech_list[color]
+	var/picked_name = pick("Gallium", "Indium", "Thallium", "Bismuth", "Aluminium", "Mercury", "Iron", "Silver", "Zinc", "Titanium", "Chromium", "Nickel", "Platinum", "Tellurium", "Palladium", "Rhodium", "Cobalt", "Osmium", "Tungsten", "Iridium")
+
+	G.name = "[picked_name] [color]"
+	G.real_name = "[picked_name] [color]"
+	G.icon_living = "[theme][color]"
+	G.icon_state = "[theme][color]"
+	G.icon_dead = "[theme][color]"
+
+	to_chat(user, "[G.tech_fluff_string].")
```
Comment (SamHPurp) [inline code/game/gamemodes/miniantags/guardian/guardian.dm]: and again

File: code/game/gamemodes/miniantags/guardian/guardian.dm [hunk 8]
```diff
@@ -435,6 +406,34 @@
...
+	"Iris" = "#2CF6B8", \
+	"Petunia" = "#51A9D4", \
+	"Violet" = "#8A347C", \
+	"Lilac" = "#C7A0F6", \
+	"Orchid" = "#F62CF5")
+
+	var/color = pick(bio_list) //technically not colors, just using the same flowers as tech currerntly
+	G.name_color = bio_list[color]
+	var/picked_name = pick("brood", "hive", "nest")
+	to_chat(user, "[G.bio_fluff_string].")
```
Comment (SamHPurp) [inline code/game/gamemodes/miniantags/guardian/guardian.dm]: ahem

File: code/game/gamemodes/miniantags/guardian/guardian.dm [hunk 9]
```diff
@@ -435,6 +406,34 @@
...
+
+	G.name = "[color] [picked_name]"
+	G.real_name = "[color] [picked_name]"
+	G.icon_living = "[theme][color]"
+	G.icon_state = "[theme][color]"
+	G.icon_dead = "[theme][color]"
+
+	to_chat(user, "[G.bio_fluff_string].")
```
Comment (SamHPurp) [inline code/game/gamemodes/miniantags/guardian/guardian.dm]: yahoo

File: code/game/gamemodes/miniantags/guardian/guardian.dm (hunk 6 above)
Comment (farie82) [reply code/game/gamemodes/miniantags/guardian/guardian.dm]: Nah the guardian has their own version depending on the type

Comment (SamHPurp) [reply code/game/gamemodes/miniantags/guardian/guardian.dm]: Oh its telling them what they got. Right, okay.

File: code/game/gamemodes/miniantags/guardian/guardian.dm [hunk 10]
```diff
@@ -271,19 +277,20 @@
 	if(candidates.len)
 		theghost = pick(candidates)
 		spawn_guardian(user, theghost.key)
+		desc += "\n[used_message]"
```
Comment (variableundefined) [inline code/game/gamemodes/miniantags/guardian/guardian.dm]: Would it not be more accurate to add a to_chat (with span class) in the examine proc to show whether it is used instead?

**Comment by farie82** [comment]
weird merge conflict with my own code. Always nice

**Pull request merged**